### DISCUSSION TITLE
Run Local Copy w/o Domain

### DISCUSSION THREAD

**Discussion by ToS0** [caprover/caprover#2051] [Q&A] [answered]
Hi, I just downloaded my working CR VM from my VPS provider to run it for testing and backup purposes locally. However, in the browser I only get the blue 'Nothing here yet :/' message.

I suppose I need to change some Nginx confs to be able to access via an IP instead of the URL, disable SSL, etc., but wasn't able to find the right place of the settings and/or there correct values.

Thanks a lot!

**Comment by githubsaturn** [comment]
Not sure what you mean by "local copy". But you can install CapRover locally, but it requires some advanced DNS configurations etc.

https://caprover.com/docs/run-locally.html

**Comment by ToS0** [comment]
Yes, I did read this, but it wasn't useful for my case, I guess. I don't want to install a new CR instance, as it is an already installed VM with some apps on it, running for years. Now I downloaded it from my provider to a KVM host on my local home network but don't get it up completely

**Reply by githubsaturn** [reply]
You just need to point the same DNS to the IP address of caprover in your local network. 

For example, you need to point *.root.domain.com to your local IP address. root.domain.com must be the same domain name that you used in the past.

**Reply by ToS0** [reply]
Thank you - I see, that would be a a general workaround which should work.

However, it doesn't for some reason. CR container says: health check for domain failed, although all domains point to the local IP (via hosts file on the server as well as the client). Container spin up and fail faster then I can copy the logs. Let'sEncrypt tries to mess with my domain (which is still live on the original VM at the VPS). Maybe I have to setup more, like a full DNS server, with reverse lookup etc.?

I feel the setup of CR is tied very much to the domains. I was hoping to just get pointed to the right config files, where I can change the domain, to a local one, or even a plain IP. But it seems this is not possible currently. Technically it shouldn't be difficult, but it wasn't probably a use case for you so far. Correct me if I am wrong. No big deal, the original CR VM is stable for 3 years now :)

**Reply by githubsaturn** [reply]
Yea exactly. You can disable healthcheck by [running this](https://caprover.com/docs/troubleshooting.html#domain-verification-failed---error-1107). But I'm not sure if that completely resolve your issue, or you'll hit another wall.

**Comment by ToS0** [comment, marked as answer]
Yes, this did also not help, but I found another way to restore and access the container data from the CR cloud instance to a local machine. Just for reference:

I installed a new VM and there I installed the app containers via docker-compose with the original [.yml](https://github.com/caprover/one-click-apps/blob/master/public/v4/apps/nextcloud.yml) file, replacing all the CR variables with actual values. Then I stopped the container in order to rsync the cloud data of the online CR containers down to the local containers. With a few tweaks, I could use my new containers (here Nextcloud and MariaDB) with my original cloud data.

Not as easy and identical as just copying the complete CR VM from cloud to local, but at least I have a working, similar copy running on site.

Thank you for your work and support!
